File: core/node.py
class Node:
    def __init__(self, name, x, y):
        self.name      = name
        self.x         = x
        self.y         = y
        self.neighbors = []   # (node, distance, pollution, traffic)

        self.g           = float('inf')
        self.h           = 0
        self.f           = float('inf')
        self.parent      = None
        self.state       = 'unvisited'
        self.pollution_g = 0.0

    def add_neighbor(self, neighbor_node, distance,
                     pollution=5.0, traffic=1.0):
        self.neighbors.append((neighbor_node, distance, pollution, traffic))

    def reset(self):
        self.g           = float('inf')
        self.h           = 0
        self.f           = float('inf')
        self.parent      = None
        self.state       = 'unvisited'
        self.pollution_g = 0.0

    def __lt__(self, other):
        return self.f < other.f

    def __repr__(self):
        return f"Node({self.name})"

File: frontend/src/core/Node.js
export class Node {
  constructor(name, x, y) {
    this.name = name;
    this.x = x;
    this.y = y;
    this.neighbors = []; // { node, distance, pollution, traffic }

    this.g = Infinity;
    this.h = 0;
    this.f = Infinity;
    this.parent = null;
    this.state = 'unvisited';
    this.pollutionG = 0;
  }

  addNeighbor(node, distance, pollution = 5.0, traffic = 1.0) {
    this.neighbors.push({ node, distance, pollution, traffic });
  }

  reset() {
    this.g = Infinity;
    this.h = 0;
    this.f = Infinity;
    this.parent = null;
    this.state = 'unvisited';
    this.pollutionG = 0;
  }
}
